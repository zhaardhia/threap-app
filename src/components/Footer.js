import React from 'react';
import PropTypes from 'prop-types';

const Footer = ({ year, link, developer }) => (
  <footer>
    <p className="text-center my-10">
      {year}
      {' '}
      -
      {' '}
      <a href={link} className="underline">
        &copy;
        {' '}
        {developer}
      </a>
    </p>
  </footer>
);

Footer.propTypes = {
  year: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  developer: PropTypes.string.isRequired,
};

export default Footer;
