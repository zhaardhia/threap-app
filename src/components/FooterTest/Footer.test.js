/**
 * test scenario for Footer
 *
 * - Footer component
 *  - should render Footer component successfully
 *
 */
import { render } from '@testing-library/react';
import Footer from '../Footer';
import '@testing-library/jest-dom';

describe('Footer component', () => {
  it('should render Footer component successfully', async () => {
    const { container } = render(<Footer
      year="2022"
      link="https://github.com/zhaardhia"
      developer="zhaardhia" />);
    expect(container).toMatchSnapshot();
  });
});
