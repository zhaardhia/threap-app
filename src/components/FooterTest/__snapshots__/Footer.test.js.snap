// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render Footer component successfully 1`] = `
<div>
  <footer>
    <p
      class="text-center my-10"
    >
      2022
       
      -
       
      <a
        class="underline"
        href="https://github.com/zhaardhia"
      >
        zhaardhia
      </a>
    </p>
  </footer>
</div>
`;
